wled00/FX.h

```diff
@@ -903,6 +917,7 @@ class WS2812FX {  // 96 bytes
     uint16_t _length;
     uint8_t  _brightness;
     uint16_t _transitionDur;
+    std::vector<uint32_t> _buffer;
```

Comment:
Why did you choose to allocate *segment* color buffer at strip level? I'm also curious why vector?

IMO a much better approach would be `static uint32_t *_buffer;` in `Segment` class with regular `malloc()/free()` if you wanted to keep only one buffer for all segments (as they are not used at the same time). The size would still need to be the size of largest possible segment which is `strip.getLengthTotal()`.

Reply:
I agree that we only need the buffer for the largest segment. I was just wondering what typically happens while WLED starts up. I'm imagining that on startup, WLED will initialize a single segment that contains all LEDs. Then as the boot preset it loaded, the largest segment might be smaller than that. Is that true?

Either way, personally I use many of my light strips sometimes with multiple segments and sometimes with a single segment. So quite often, the largest segment is the entire strip for me. The reason why I went with a single buffer at the strip level is that I only ever have to reallocate that buffer if the strip configuration changes, which should happen much more rarely than changes to segments.

The tradeoff here is using the largest amount of memory all the time vs. potentially having to reallocate the buffer, which can lead to memory fragmentation and also costs CPU time. I also wonder if we should reallocate every time the largest segment changes? Or only if the largest segment size *increases*? I'm new to this platform, so I don't know what it is usually optimized for, so happy to go with either suggestion.

As per `vector` vs. `new uint_32[...]` vs `malloc()` ... I don't really care other than that it's harder to introduce memory bugs with vectors. I realize there's a small overhead of two integers (capacity and size). OTOH, there are already 5 other vectors in the `WS2812FX` class. Curious, any specific reason you think that in this particular case, `malloc()` is preferable?

Reply:
Yes, the approach is correct, but why in `WS2812FX` class? I think it fits better in `Segment` class as a `static` as it is only used for segment functions.

The purpose of vectors used in `WS2812FX` is to simplify dynamic *expansion* while maintaining simplicity of arrays. Unless heap becomes low, memory is never deallocated from vector. In your case you should only allocate memory once and never release or reallocate it.